### ISSUE TITLE
CA can not scale other IG group if spot unavailable 

### ISSUE THREAD

**Issue by yogeshkk** [kubernetes/autoscaler#3042]
Hello Team,

I am trying to fix CA in our environment. We have multiple Spot and on-demand IG in our environment. I am trying failover to on-demand or rest spot when there is no capacity in one IG. 

I am running the 1.12.8 on KOPS in AWS. 

We have 3 spot-large.k8s.example.com running in the environment so I have changed min price to .01 so no new spot should be created but an instant of a spinning node from other IG CA will try to spin node from spot large even after it marked as unhealthy. 

Logs attached once IG is marked as unhealthy. 

```
I0413 07:00:57.729108       1 static_autoscaler.go:121] Starting main loop
W0413 07:00:57.774140       1 aws_manager.go:204] Found multiple availability zones, using eu-west-1b
W0413 07:00:57.774790       1 aws_manager.go:204] Found multiple availability zones, using eu-west-1b
W0413 07:00:57.775908       1 aws_manager.go:204] Found multiple availability zones, using eu-west-1b
W0413 07:00:57.776885       1 aws_manager.go:204] Found multiple availability zones, using eu-west-1b
W0413 07:00:57.777571       1 aws_manager.go:204] Found multiple availability zones, using eu-west-1b
W0413 07:00:57.779114       1 clusterstate.go:507] Failed to get nodegroup for aws:///eu-west-1b/i-placeholder-spot-large.k8s.example.com-0: Wrong id: expected format aws:///<zone>/<name>, got aws:///eu-west-1b/i-placeholder-spot-large.k8s.example.com-0
W0413 07:00:57.779132       1 clusterstate.go:507] Failed to get nodegroup for aws:///eu-west-1b/i-placeholder-spot-large.k8s.example.com-1: Wrong id: expected format aws:///<zone>/<name>, got aws:///eu-west-1b/i-placeholder-spot-large.k8s.example.com-1
W0413 07:00:57.779142       1 clusterstate.go:507] Failed to get nodegroup for aws:///eu-west-1b/i-placeholder-spot-large.k8s.example.com-2: Wrong id: expected format aws:///<zone>/<name>, got aws:///eu-west-1b/i-placeholder-spot-large.k8s.example.com-2
W0413 07:00:57.779157       1 clusterstate.go:542] Readiness for node group spot-large.k8s.example.com not found
I0413 07:00:57.779192       1 static_autoscaler.go:168] 3 unregistered nodes present
I0413 07:00:57.779243       1 static_autoscaler.go:253] Filtering out schedulables
I0413 07:00:57.779593       1 static_autoscaler.go:263] No schedulable pods
I0413 07:00:57.779610       1 scale_up.go:264] Pod kube-addons/namespace-configuration-controller-75488f946-5fxj4 is unschedulable
I0413 07:00:57.779618       1 scale_up.go:264] Pod default/ubuntu-7dd49546bf-rwcnv is unschedulable
I0413 07:00:57.779624       1 scale_up.go:264] Pod kube-addons/dex-66d54446cb-nm5k5 is unschedulable
I0413 07:00:57.779630       1 scale_up.go:264] Pod ingress-nginx-public/nginx-ingress-controller-c595b6f7b-tppsg is unschedulable
I0413 07:00:57.779636       1 scale_up.go:264] Pod kube-addons/aws-autoscaling-exporter-77ff5cdb-nddgr is unschedulable
I0413 07:00:57.779642       1 scale_up.go:264] Pod ingress-nginx-internal/nginx-ingress-controller-5cc89f6c64-x9kbq is unschedulable
I0413 07:00:57.779648       1 scale_up.go:264] Pod monitoring/grafana-5984757469-rkfp7 is unschedulable
I0413 07:00:57.779654       1 scale_up.go:264] Pod default/ubuntu-7dd49546bf-42cgc is unschedulable
I0413 07:00:57.779659       1 scale_up.go:264] Pod default/ubuntu-7dd49546bf-mdnks is unschedulable
I0413 07:00:57.779721       1 scale_up.go:301] Upcoming 3 nodes
I0413 07:00:57.783775       1 scale_up.go:404] No need for any nodes in on-demand-large.k8s.example.com
I0413 07:00:57.783813       1 scale_up.go:336] Skipping node group on-demand-medium.k8s.example.com - max size reached
I0413 07:00:57.786889       1 scale_up.go:404] No need for any nodes in on-demand-xlarge.k8s.example.com
W0413 07:00:57.786908       1 clusterstate.go:324] Failed to find readiness information for spot-large.k8s.example.com
W0413 07:00:57.786914       1 clusterstate.go:324] Failed to find readiness information for spot-large.k8s.example.com
W0413 07:00:57.786923       1 scale_up.go:320] Node group spot-large.k8s.example.com is not ready for scaleup - unhealthy
I0413 07:00:57.786930       1 scale_up.go:336] Skipping node group spot-medium.k8s.example.com - max size reached
I0413 07:00:57.791445       1 scale_up.go:404] No need for any nodes in spot-xlarge.k8s.example.com
I0413 07:00:57.791470       1 scale_up.go:412] No expansion options
I0413 07:00:57.791494       1 static_autoscaler.go:308] Calculating unneeded nodes
I0413 07:00:57.791537       1 utils.go:500] Skipping ip-10-40-2-217.eu-west-1.compute.internal - no node group config
I0413 07:00:57.791556       1 utils.go:500] Skipping ip-10-40-1-89.eu-west-1.compute.internal - no node group config
I0413 07:00:57.791692       1 scale_down.go:402] Node ip-10-40-3-15.eu-west-1.compute.internal - utilization 0.972937
I0413 07:00:57.791705       1 scale_down.go:406] Node ip-10-40-3-15.eu-west-1.compute.internal is not suitable for removal - utilization too big (0.972937)
I0413 07:00:57.791716       1 scale_down.go:402] Node ip-10-40-2-214.eu-west-1.compute.internal - utilization 0.977546
I0413 07:00:57.791723       1 scale_down.go:406] Node ip-10-40-2-214.eu-west-1.compute.internal is not suitable for removal - utilization too big (0.977546)
I0413 07:00:57.791834       1 static_autoscaler.go:334] Scale down status: unneededOnly=true lastScaleUpTime=2020-04-13 06:53:43.783742619 +0000 UTC m=+26.411167251 lastScaleDownDeleteTime=2020-04-13 06:53:43.78374393 +0000 UTC m=+26.411168563 lastScaleDownFailTime=2020-04-13 06:53:43.783745233 +0000 UTC m=+26.411169872 scaleDownForbidden=false isDeleteInProgress=false
W0413 07:00:57.791874       1 clusterstate.go:324] Failed to find readiness information for spot-large.k8s.example.com
W0413 07:00:57.791884       1 clusterstate.go:380] Failed to find readiness information for spot-large.k8s.example.com
W0413 07:00:57.791890       1 clusterstate.go:324] Failed to find readiness information for spot-large.k8s.example.com
I0413 07:00:57.806923       1 leaderelection.go:227] successfully renewed lease kube-addons/cluster-autoscaler
I0413 07:00:59.421104       1 reflector.go:357] k8s.io/client-go/informers/factory.go:131: Watch close - *v1.PersistentVolumeClaim total 0 items received
I0413 07:00:59.841312       1 leaderelection.go:227] successfully renewed lease kube-addons/cluster-autoscaler
I0413 07:01:01.875432       1 leaderelection.go:227] successfully renewed lease kube-addons/cluster-autoscaler
I0413 07:01:03.896896       1 leaderelection.go:227] successfully renewed lease kube-addons/cluster-autoscaler
I0413 07:01:05.926116       1 leaderelection.go:227] successfully renewed lease kube-addons/cluster-autoscaler
I0413 07:01:07.827308       1 static_autoscaler.go:121] Starting main loop
```

We do have the capacity in on-demand-large.k8s.example.com or spot-xlarge.k8s.example.com still it says No need of node

```
I0413 07:00:57.783775       1 scale_up.go:404] No need for any nodes in on-demand-large.k8s.example.com
I0413 07:00:57.791445       1 scale_up.go:404] No need for any nodes in spot-xlarge.k8s.example.com

```

When I open the AWS console I found that in IG spot-large.k8s.example.com desired capacity was set to 3. I have changed that to zero and then one node was spin up from on-demand-large.k8s.example.com.

```
I0413 07:04:33.706098       1 static_autoscaler.go:121] Starting main loop
W0413 07:04:33.789973       1 aws_manager.go:204] Found multiple availability zones, using eu-west-1b
W0413 07:04:33.790690       1 aws_manager.go:204] Found multiple availability zones, using eu-west-1b
W0413 07:04:33.791244       1 aws_manager.go:204] Found multiple availability zones, using eu-west-1b
W0413 07:04:33.791760       1 aws_manager.go:204] Found multiple availability zones, using eu-west-1b
W0413 07:04:33.792479       1 aws_manager.go:204] Found multiple availability zones, using eu-west-1b
I0413 07:04:34.032968       1 auto_scaling_groups.go:344] Regenerating instance to ASG map for ASGs: [on-demand-large.k8s.example.com on-demand-medium.k8s.example.com on-demand-xlarge.k8s.example.com spot-large.k8s.example.com spot-medium.k8s.example.com spot-xlarge.k8s.example.com]
I0413 07:04:34.112555       1 auto_scaling_groups.go:106] Updating ASG spot-large.k8s.example.com
I0413 07:04:34.126902       1 aws_manager.go:158] Refreshed ASG list, next refresh after 2020-04-13 07:05:34.126890778 +0000 UTC m=+736.754315873
I0413 07:04:34.127217       1 static_autoscaler.go:253] Filtering out schedulables
I0413 07:04:34.127668       1 static_autoscaler.go:263] No schedulable pods
I0413 07:04:34.127695       1 scale_up.go:264] Pod default/ubuntu-7dd49546bf-42cgc is unschedulable
I0413 07:04:34.127703       1 scale_up.go:264] Pod default/ubuntu-7dd49546bf-mdnks is unschedulable
I0413 07:04:34.127713       1 scale_up.go:264] Pod kube-addons/aws-autoscaling-exporter-77ff5cdb-nddgr is unschedulable
I0413 07:04:34.127719       1 scale_up.go:264] Pod ingress-nginx-internal/nginx-ingress-controller-5cc89f6c64-x9kbq is unschedulable
I0413 07:04:34.127725       1 scale_up.go:264] Pod monitoring/grafana-5984757469-rkfp7 is unschedulable
I0413 07:04:34.127730       1 scale_up.go:264] Pod ingress-nginx-public/nginx-ingress-controller-c595b6f7b-tppsg is unschedulable
I0413 07:04:34.127740       1 scale_up.go:264] Pod kube-addons/namespace-configuration-controller-75488f946-5fxj4 is unschedulable
I0413 07:04:34.127746       1 scale_up.go:264] Pod default/ubuntu-7dd49546bf-rwcnv is unschedulable
I0413 07:04:34.127752       1 scale_up.go:264] Pod kube-addons/dex-66d54446cb-nm5k5 is unschedulable
I0413 07:04:34.127795       1 scale_up.go:301] Upcoming 0 nodes
I0413 07:04:34.129637       1 scale_up.go:336] Skipping node group on-demand-medium.k8s.example.com - max size reached
I0413 07:04:34.172786       1 scale_up.go:336] Skipping node group spot-medium.k8s.example.com - max size reached
I0413 07:04:34.192770       1 waste.go:57] Expanding Node Group on-demand-large.k8s.example.com would waste 91.75% CPU, 47.71% Memory, 69.73% Blended
I0413 07:04:34.192977       1 waste.go:57] Expanding Node Group on-demand-xlarge.k8s.example.com would waste 95.88% CPU, 73.85% Memory, 84.86% Blended
I0413 07:04:34.192989       1 waste.go:57] Expanding Node Group spot-large.k8s.example.com would waste 95.88% CPU, 47.71% Memory, 71.79% Blended
I0413 07:04:34.193018       1 waste.go:57] Expanding Node Group spot-xlarge.k8s.example.com would waste 95.88% CPU, 73.85% Memory, 84.86% Blended
I0413 07:04:34.193034       1 scale_up.go:419] Best option to resize: on-demand-large.k8s.example.com
I0413 07:04:34.193043       1 scale_up.go:423] Estimated 1 nodes needed in on-demand-large.k8s.example.com
I0413 07:04:34.193070       1 scale_up.go:502] Final scale-up plan: [{on-demand-large.k8s.example.com 0->1 (max: 5)}]
I0413 07:04:34.193093       1 scale_up.go:588] Scale-up: setting group on-demand-large.k8s.example.com size to 1
I0413 07:04:34.193368       1 auto_scaling_groups.go:211] Setting asg on-demand-large.k8s.example.com size to 1
I0413 07:04:34.240048       1 factory.go:33] Event(v1.ObjectReference{Kind:"ConfigMap", Namespace:"kube-addons", Name:"cluster-autoscaler-status", UID:"80f4ba36-7d53-11ea-820a-0a2151a7b874", APIVersion:"v1", ResourceVersion:"18707775", FieldPath:""}): type: 'Normal' reason: 'ScaledUpGroup' Scale-up: setting group on-demand-large.k8s.example.com size to 1
I0413 07:04:34.380708       1 factory.go:33] Event(v1.ObjectReference{Kind:"Pod", Namespace:"kube-addons", Name:"dex-66d54446cb-nm5k5", UID:"4ed0e4b1-7d51-11ea-820a-0a2151a7b874", APIVersion:"v1", ResourceVersion:"18707523", FieldPath:""}): type: 'Normal' reason: 'TriggeredScaleUp' pod triggered scale-up: [{on-demand-large.k8s.example.com 0->1 (max: 5)}]
I0413 07:04:34.380743       1 factory.go:33] Event(v1.ObjectReference{Kind:"ConfigMap", Namespace:"kube-addons", Name:"cluster-autoscaler-status", UID:"80f4ba36-7d53-11ea-820a-0a2151a7b874", APIVersion:"v1", ResourceVersion:"18707775", FieldPath:""}): type: 'Normal' reason: 'ScaledUpGroup' Scale-up: group on-demand-large.k8s.example.com size set to 1
I0413 07:04:34.380758       1 factory.go:33] Event(v1.ObjectReference{Kind:"Pod", Namespace:"default", Name:"ubuntu-7dd49546bf-42cgc", UID:"3e26ac34-7d51-11ea-820a-0a2151a7b874", APIVersion:"v1", ResourceVersion:"18707527", FieldPath:""}): type: 'Normal' reason: 'TriggeredScaleUp' pod triggered scale-up: [{on-demand-large.k8s.example.com 0->1 (max: 5)}]
I0413 07:04:34.380770       1 factory.go:33] Event(v1.ObjectReference{Kind:"Pod", Namespace:"default", Name:"ubuntu-7dd49546bf-mdnks", UID:"3e5b99e8-7d51-11ea-820a-0a2151a7b874", APIVersion:"v1", ResourceVersion:"18707807", FieldPath:""}): type: 'Normal' reason: 'TriggeredScaleUp' pod triggered scale-up: [{on-demand-large.k8s.example.com 0->1 (max: 5)}]
I0413 07:04:34.380782       1 factory.go:33] Event(v1.ObjectReference{Kind:"Pod", Namespace:"kube-addons", Name:"aws-autoscaling-exporter-77ff5cdb-nddgr", UID:"47f9d278-7d51-11ea-820a-0a2151a7b874", APIVersion:"v1", ResourceVersion:"18707510", FieldPath:""}): type: 'Normal' reason: 'TriggeredScaleUp' pod triggered scale-up: [{on-demand-large.k8s.example.com 0->1 (max: 5)}]
I0413 07:04:34.380794       1 factory.go:33] Event(v1.ObjectReference{Kind:"Pod", Namespace:"ingress-nginx-internal", Name:"nginx-ingress-controller-5cc89f6c64-x9kbq", UID:"4a128cb8-7d51-11ea-820a-0a2151a7b874", APIVersion:"v1", ResourceVersion:"18707514", FieldPath:""}): type: 'Normal' reason: 'TriggeredScaleUp' pod triggered scale-up: [{on-demand-large.k8s.example.com 0->1 (max: 5)}]
I0413 07:04:34.380806       1 factory.go:33] Event(v1.ObjectReference{Kind:"Pod", Namespace:"kube-addons", Name:"namespace-configuration-controller-75488f946-5fxj4", UID:"4f642dc1-7d51-11ea-820a-0a2151a7b874", APIVersion:"v1", ResourceVersion:"18707710", FieldPath:""}): type: 'Normal' reason: 'TriggeredScaleUp' pod triggered scale-up: [{on-demand-large.k8s.example.com 0->1 (max: 5)}]
I0413 07:04:34.380818       1 factory.go:33] Event(v1.ObjectReference{Kind:"Pod", Namespace:"default", Name:"ubuntu-7dd49546bf-rwcnv", UID:"3e5bd5b4-7d51-11ea-820a-0a2151a7b874", APIVersion:"v1", ResourceVersion:"18707806", FieldPath:""}): type: 'Normal' reason: 'TriggeredScaleUp' pod triggered scale-up: [{on-demand-large.k8s.example.com 0->1 (max: 5)}]
I0413 07:04:34.380830       1 factory.go:33] Event(v1.ObjectReference{Kind:"Pod", Namespace:"monitoring", Name:"grafana-5984757469-rkfp7", UID:"51dd6a53-7d51-11ea-820a-0a2151a7b874", APIVersion:"v1", ResourceVersion:"18707611", FieldPath:""}): type: 'Normal' reason: 'TriggeredScaleUp' pod triggered scale-up: [{on-demand-large.k8s.example.com 0->1 (max: 5)}]
I0413 07:04:34.380842       1 factory.go:33] Event(v1.ObjectReference{Kind:"Pod", Namespace:"ingress-nginx-public", Name:"nginx-ingress-controller-c595b6f7b-tppsg", UID:"463511e1-7d51-11ea-820a-0a2151a7b874", APIVersion:"v1", ResourceVersion:"18707526", FieldPath:""}): type: 'Normal' reason: 'TriggeredScaleUp' pod triggered scale-up: [{on-demand-large.k8s.example.com 0->1 (max: 5)}]
I0413 07:04:34.670180       1 leaderelection.go:227] successfully renewed lease kube-addons/cluster-autoscaler
I0413 07:04:36.798953       1 leaderelection.go:227] successfully renewed lease kube-addons/cluster-autoscaler
I0413 07:04:38.989767       1 leaderelection.go:227] successfully renewed lease kube-addons/cluster-autoscaler
I0413 07:04:41.362349       1 leaderelection.go:227] successfully renewed lease kube-addons/cluster-autoscaler
I0413 07:04:43.582750       1 leaderelection.go:227] successfully renewed lease kube-addons/cluster-autoscaler
I0413 07:04:44.563669       1 static_autoscaler.go:121] Starting main loop
```

I am not sure about the internal but should CA change the desired node count in IG to zero after marking the node group as unhealthy? 

Currently, it did not happen and CA infinitely try to add a node from unhealthy IG which should not be the case

Let me know if I am doing something wrong.

**Comment by yogeshkk** [comment]
Hi,

I just wanted some input it is a bug or misconfiguration. One can easily reproduce it. If you require any more information please let me know.

**Comment by gjtempleton** [comment]
Hi @yogeshkk 

It looks like you're running into this version of the CA not handling spot instances well. How long did you wait after the spot requests being created before manually making the changes to ASG desired count? It'll wait for the length of `max-node-provision-time` of the placeholder instances being in progress and not turning into real nodes before it scales the ASG they're in back down.

**Comment by MaciekPytel** [comment]
My guess is with enough spot ASGs it can never scale-up. It will try one spot, wait for `max-node-provision-time`, backoff that ASG and retry scale-up in a different one. If there are enough spot ASGs the first one will go off backoff before the scale-up in the last one times out. Or am I missing something here?

I guess a FR here would be to have AWS provider poll infrastructure and update status of placeholder, so that scale-ups fail more quickly. This is already how GCE provider does it, but I don't know enough about AWS to say how difficult it would be to add this (MIGs expose create errors in a way that makes them easy to consume for CA, I don't know how ASGs do it).

**Comment by yogeshkk** [comment]
@gjtempleton,

I was using the old version so I have a bump to the latest available for 1.12 which is 1.12.8.  I think max-node-provision-time is 15 min. I have waited around the half-hour and then gave up/

From logs, you can see it marking ASG as unhealthy still it does not try to scale from other ASG and use this same.

`W0413 07:00:57.786923       1 scale_up.go:320] Node group spot-large.k8s.example.com is not ready for scaleup - unhealthy`

I was curious to this error I don't know if this contributes to anything. 
```
W0413 07:00:57.779114       1 clusterstate.go:507] Failed to get nodegroup for aws:///eu-west-1b/i-placeholder-spot-large.k8s.example.com-0: Wrong id: expected format aws:///<zone>/<name>, got aws:///eu-west-1b/i-placeholder-spot-large.k8s.example.com-0
W0413 07:00:57.779132       1 clusterstate.go:507] Failed to get nodegroup for aws:///eu-west-1b/i-placeholder-spot-large.k8s.example.com-1: Wrong id: expected format aws:///<zone>/<name>, got aws:///eu-west-1b/i-placeholder-spot-large.k8s.example.com-1
W0413 07:00:57.779142       1 clusterstate.go:507] Failed to get nodegroup for aws:///eu-west-1b/i-placeholder-spot-large.k8s.example.com-2: Wrong id: expected format aws:///<zone>/<name>, got aws:///eu-west-1b/i-placeholder-spot-large.k8s.example.com-2
```

**Comment by gjtempleton** [comment]
Unfortunately it looks like you've run into [this bug](https://github.com/kubernetes/autoscaler/issue/2378) due to the naming of your ASGs. 

The fix for this was only cherry-picked back as far as release branch 1.13 so didn't make it into 1.12 despite the new placeholder instance creation logic making it into 1.12. This means the CA can never successfully "delete" your placeholder instances and scale the unhealthy ASG back down, so it never actually tries the other ASGs as it sees the 3 nodes as upcoming eternally.

**Comment by yogeshkk** [comment]
Thanks a lot, @gjtempleton for pointing out the bug and helping on the issue. 

Just curious by any chance, you are planning to cherry-pick for 1.12?  otherwise, we have to go a similar path as you did for your 1.12 cluster.

**Comment by gjtempleton** [comment]
At the time I raised the issue and PR/Cherry-Picks 1.12 had fallen far enough out of support as 1.16 had been recently been released that I decided not to cherry-pick it back, given that would be 2 minor versions back from the oldest supported k8s version.

Given that 1.18 has now been released 1.12 is now 4 releases back from the oldest supported k8s version I'm not sure how valuable it would be to cherry-pick it (though it would obviously resolve your problem). I don't cut the releases so I'm unsure of the chances of a 1.12 release being cut after all this time even if it was picked back.

The other option is you could try running with the 1.13 release which includes the patch (although the usual warning around potential issues with scheduling assumptions apply).

**Comment by yogeshkk** [comment]
Fair enough we are not fast at adoption newer version so this is on us. No issues 

Btw many thanks @gjtempleton atleast we have idea how to move ahead. 
I will be closed the issue.

**Comment by yogeshkk** [comment]
Just in case anyone facing the same issue. I have creaked fork of repo https://github.com/yogeshkk/autoscaler which contains all steps and the docker image is yogeshkunjir/clusterautoscaler:v1.12.8.patch-fix for handy. 

I have checked the image is working as expected so you can use my image. I do recommend creating your own images as using steps provided in Github repo.

**Comment by nitrag** [comment]
Is this bug patched in EKS 1.15+? I ask because I'm running into this issue. I try and scale up a spot instance and 10 minutes later it is creating a placeholder instance. `(FLAG: --max-node-provision-time="5m0s")` is ignored. Should I create a separate github issue for this?

```
2020-09-02T11:22:40.246-04:00
{"log":"I0902 15:22:40.246183 1 scale_up.go:699] Scale-up: setting group pv-platform-prod-worker-gpu-accelerated-p2-xlarge-spot size to 1


2020-09-02T11:32:08.458-04:00
Failed to find readiness information for pv-platform-prod-worker-gpu-accelerated-p2-xlarge-spot
Failed to find readiness information for pv-platform-prod-worker-gpu-accelerated-p2-xlarge-spot
Failed to find readiness information for pv-platform-prod-worker-gpu-accelerated-p2-xlarge-spot
Failed to find readiness information for pv-platform-prod-worker-gpu-accelerated-p2-xlarge-spot
Failed to find readiness information for pv-platform-prod-worker-gpu-accelerated-p2-xlarge-spot
Failed to find readiness information for pv-platform-prod-worker-gpu-accelerated-p2-xlarge-spot
2020-09-02T11:33:20.650-04:00


2020-09-02T11:33:10.353-04:00
{"log":"W0902 15:33:10.353075 1 scale_up.go:326] Node group pv-platform-prod-worker-gpu-accelerated-p2-xlarge-spot is not ready for scaleup - unhealthy


2020-09-02T11:33:10.833-04:00
{"log":"I0902 15:33:10.831490 1 auto_scaling_groups.go:407] Instance group pv-platform-prod-worker-gpu-accelerated-p2-xlarge-spot has only 0 instances created while requested count is 1. Creating placeholder instance with ID i-placeholder-pv-platform-prod-worker-gpu-accelerated-p2-xlarge-spot-0
```

Expected behavior: 5 minutes after scale up fails to launch a spot instance, it should fallback to on demand (lower priority ASG (we use priority expander)).
